lib/usb/usb_standard.c

```diff
@@ -301,11 +313,17 @@ static int usb_standard_set_interface(usbd_device *usbd_dev,
 				      struct usb_setup_data *req,
 				      uint8_t **buf, uint16_t *len)
 {
-	const struct usb_config_descriptor *cfx = &usbd_dev->config[usbd_dev->current_config - 1];
+	const struct usb_config_descriptor *cfx;
 	const struct usb_interface *iface;
 
 	(void)buf;
 
+	if (! usbd_dev->current_config) {
+		return USBD_REQ_NOTSUPP;
+	}
```

Comment:
Is it really necessary to add this check?  It seems like paranoia.  Can you ever set_interface without having a config?